libsolidity/analysis/ContractLevelChecker.cpp

```diff
@@ -510,3 +737,96 @@ void ContractLevelChecker::checkBaseABICompatibility(ContractDefinition const& _
 		);
 
 }
+
+void ContractLevelChecker::checkAmbiguousOverrides(ContractDefinition const& _contract) const
+{
+	vector<FunctionInfo> baseFuncs;
+
+	for (ContractDefinition const* c: resolveBaseContracts(_contract))
+		for (pair<string, FunctionInfo> const& funcPair: getOverridableFunctions(c))
+		{
+			auto baseFunc = funcPair.second.funcDef;
```

Comment:
It takes me 3 seconds to find out the type of `baseFunc` if I am not used to the code.